src/mqueryfront/src/QueryMonaco.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import Editor, { monaco } from "@monaco-editor/react";
+import YARA from "./yara-lang";
+
+class QueryMonaco extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            rawYara: props.rawYara,
```

Comment:
Shouldn't we use `getDerivedStateFromProps` here? (https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) It could be glitchy in case rawYara passed as property has been changed without remounting the component.

Reply:
This suggestion contradicts the suggestions you mentioned on your second review. Thus, I feel like this one can be put aside ^

Reply:
That's true, it should be closed :)